File: autoburn.android/app/src/main/java/com/autoburn/msghandler/RxMsgHandlerBase.java
package com.autoburn.msghandler;

import android.os.AsyncTask;
/**
 * Created by chowhsienwu on 2017/5/4.
 */
public class RxMsgHandlerBase  extends  AsyncTask<String, Void, Void> implements MsgBase{

    public static final String TAG = "RxMsgHandlerBase";
    public String Type;
    public String Msg;

    public RxMsgHandlerBase(String type){
        Type = type;
    }

    /* from AsyncTask*/
    @Override
    protected Void doInBackground(String... params) {

        return null;
    }

    // to process
}

File: autoburn.pc/autoburn/msghandler/RxMsgHandlerBase.cs
using Autoburn.Manager;
using Autoburn.Net;
using Autoburn.util;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Autoburn.MsgHandler
{
    public abstract class RxMsgHandlerBase :MsgBase
    {
        private new string TAG = "RxMsgHandlerBase";

        public string Type;
        protected DeviceManager devicesmanager;
        public RxMsgHandlerBase(DeviceManager d,  string type)
        {
            devicesmanager = d;
            Type = type;
        }
        public virtual void Process() {
            D("msghandler msg " + Msg);
        }

        public string Msg;
    }
}
